slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -94,6 +94,12 @@ trait MySQLProfile extends JdbcProfile { profile =>
         else l
       }
     }
+
+    //Reference: https://github.com/slick/slick/issues/1419
+    override def createTableNamer(meta: MTable): TableNamer = new TableNamer(meta){
+      override def schema = meta.name.catalog
+      override def catalog = meta.name.schema 
```

Comment:
@trevorsibanda @lastland It seems to me that it is pretty much up to the JDBC driver to define what schema and catalog mean. I had trouble finding authoritative info on this. This SO answer seems to support that this change does indeed fix is: http://stackoverflow.com/a/5679435/467390. Since we only have one code generator but driver specific model builders, it seems that normalizing the meta data in the model builder is indeed the right thing to do. It's a bit weird that we do this cross name mapping here, but one way to think about it: The MTable, etc. api exposes access to the JDBC drivers model. The ModelBuilder is Slick's model, which is different here. So it's still confusing that schema and catalog are reversed here compared to the JDBC driver but seems like the best we got.